Lib/test/test_functools.py

```diff
@@ -2792,6 +2792,41 @@ def _(arg: int | float):
         self.assertEqual(f(1), "types.UnionType")
         self.assertEqual(f(1.0), "types.UnionType")
 
+    def test_union_None(self):
+        @functools.singledispatch
+        def f(arg):
+            return "default"
+
+        @f.register
+        def _(arg: typing.Union[str, None]):
+            return "typing.Union"
+
+        @f.register
+        def _(arg: int):
+            return "types.UnionType"
+
+        self.assertEqual(f([]), "default")
+        self.assertEqual(f(""), "typing.Union")
+        self.assertEqual(f(None), "typing.Union")
+        self.assertEqual(f(1), "types.UnionType")
+
+        @functools.singledispatch
+        def f(arg):
+            return "default"
+
+        @f.register
+        def _(arg: str):
+            return "typing.Union"
+
+        @f.register
+        def _(arg: int | None):
+            return "types.UnionType"
+
+        self.assertEqual(f([]), "default")
+        self.assertEqual(f(""), "typing.Union")
+        self.assertEqual(f(1), "types.UnionType")
+        self.assertEqual(f(None), "types.UnionType")
+
```

Comment:
Please add a test for conflicting typing.Union and types.UnionType.

Reply:
Done (along with your other suggestions)